test/process/index/IndexerTest.cpp

```diff
@@ -0,0 +1,447 @@
+
+                CHECK(storedHeader);
+                CHECK(Indexer::INDEXER_VERSION == storedHeader->indexWriterVersion);
+
+                CHECK(1 == storedEntries.size());
+                CHECK(numberOfLinesInTestFASTQ == storedLineCount);
+
+        result = TestResourcesAndFunctions::extractGZFile(fastq, extractedFastq);
+                CHECK_EQUAL(true, result);
+
+        vector<string> decompressedSourceContent = TestResourcesAndFunctions::readLinesOfFile(extractedFastq);
+
+                CHECK(TestResourcesAndFunctions::compareVectorContent(storedLines, decompressedSourceContent));
+
+        delete indexer;
+    }
+
+    TEST (TEST_CREATE_INDEX_CONCAT) {
+        TestResourcesAndFunctions res(INDEXER_SUITE_TESTS, TEST_CREATE_INDEX_SMALL);
+
+        path fastq = res.getResource(TEST_FASTQ_SMALL);
+        path concat = res.filePath("test_concat.fastq.gz");
+        path index = res.filePath("test_concat.fastq.gz.fqi");
+        path extractedFastq = res.filePath(TEST_FASTQ_SMALL);
+
+
+        int appendCount = 4;
+        bool result = TestResourcesAndFunctions::createConcatenatedFile(fastq, concat, appendCount);
+                CHECK_EQUAL(true, result);
+                CHECK(4 * file_size(fastq) == file_size(concat));
+
+        auto *indexer = new Indexer(make_shared<PathSource>(concat),
+                                    make_shared<PathSink>(index),
+                                    BlockDistanceStorageStrategy::getDefault(),
+                                    true,
+                                    false,
+                                    false,
+                                    true
+        );
+                CHECK(indexer->fulfillsPremises());  // We need to make sure things are good. Also this opens the I-Writer.
+
+        result = indexer->createIndex();
+
+        auto storedHeader = indexer->getStoredHeader();
+        auto storedEntries = indexer->getStoredEntries();
+        auto storedLines = indexer->getStoredLines();
+
+        int numberOfLinesInTestFASTQ = appendCount * 4000;
+        int storedLineCount = storedLines.size();
+                CHECK(result);
+                CHECK(exists(index));
+                CHECK(indexer->wasSuccessful());
+
+                CHECK(storedHeader);
+                CHECK_EQUAL(Indexer::INDEXER_VERSION, storedHeader->indexWriterVersion);
+
+                CHECK_EQUAL(1U, storedEntries.size());
+                CHECK(numberOfLinesInTestFASTQ == storedLineCount);
+
+        result = TestResourcesAndFunctions::extractGZFile(concat, extractedFastq);
+                CHECK_EQUAL(true, result);
+
+        vector<string> decompressedSourceContent = TestResourcesAndFunctions::readLinesOfFile(extractedFastq);
+
+                CHECK(TestResourcesAndFunctions::compareVectorContent(storedLines, decompressedSourceContent));
+
+        // Why is this a pointer? Just to get access to the file on the command line. It is written if the
```

Comment:
1. With the "pointer", do you mean `indexer`?
```suggestion
        // Why is `indexer` a pointer? Just to get access to the file on the command line. It is written if the
```
2. The pointer is "written"?
3. Should the comment not be with the declaration/definion of the pointer?